src/connector/codec/src/decoder/avro/schema.rs

```diff
@@ -124,11 +124,15 @@ fn avro_field_to_column_desc(
             )
         }
         Schema::Record(RecordSchema {
-            name: schema_name,
+            // TODO(struct): assign type name once supported.
+            name: _type_name,
             fields,
             ..
         }) => {
-            let vec_column = fields
+            // TODO(struct): since we deprecated `field_descs` in `ColumnDesc`, there's no need to
+            // traverse the fields here except for maintaining the same column id (index) as the
+            // original implementation.
```

Comment:
This is the immediate motivation that I restarted #19538 after putting #13104 on hold.

Reply:
Actually, because we always reassign the `ColumnId` in the final step of creating a source or table, there's no need to allocate a column ID here (but simply specifying `ColumnId::placeholder()`), unless we need the uniqueness property of `ColumnId` before that step.